### PR TITLE
Dm 3941 add image support to news pb component

### PR DESCRIPTION
**Pull request by PhilipDeFraties**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-3941?focusedCommentId=11739&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-11739

## Description - what does this code do?
* Adds image support to PB news component
* fixes bottom margin spacing for component

## Testing done - how did you test it/steps on how can another person can test it 
1. Add a news component to a page
2. Add an image, verify the component only persists when alt image text is submitted with image file
3. Verify the image appears above the news article text when the page renders
4. Back at page edit screen, verify an image preview now appears in the input section for the news component you added
5. Add multiple images, verify they all render above their respective text bodies
6. Verify a news component can be added without an image, verify the component renders on the page

## Screenshots, Gifs, Videos from application (if applicable)
Screenshot 2023-07-11 at 1 30 34 PM

Screenshot 2023-07-11 at 1 44 23 PM

Bonus: fixed margin spacing at bottom of component when "Load More" button is present.

Before:
Screenshot 2023-07-07 at 2 58 12 PM

After:
Screenshot 2023-07-07 at 2 58 03 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/ccwLbt94U9QGfnuFCwOXzY/Core-Designs---VA-DM?type=design&node-id=6181-38719&mode=design&t=WiSeGWLO9J7jdBKp-0

## Acceptance criteria
- [ ]

## Definition of done
- [x] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating JIRA issue
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/page/show.html.erb [hunk 1]
```diff
@@ -195,8 +195,8 @@
               nic = @news_items_components[component_index]
               @news_list_component_index = component_index + 1
             %>
-            <div class="grid-container">
-              <ul class="usa-card-group news-component-list dm-paginated-<%= component_index %>-news <%= page_narrow_classes %><%= ' margin-bottom-0' if last_component === index %>">
+            <div class="grid-container margin-bottom-5 <%= ' margin-bottom-0' if last_component === index %>">
+              <ul class="usa-card-group news-component-list dm-paginated-<%= component_index %>-news <%= page_narrow_classes %>">
```
Comment (PhilipDeFraties) [inline app/views/page/show.html.erb:199]: this is where I made an adjustment for the bottom margin, and now that I'm seeing that it works I'm wondering if I should go back and move the conditional bottom margins for the other components to be in the `grid-container` div classes as well for the sake of consistency?

Also, I noticed that the "Load More" button seems to appear when there are as few as 2 news components, however it only seems to hide components that are the 5th and beyond, is this a bug?

Comment (camillevilla) [reply app/views/page/show.html.erb:199]: re: adjusting margins, that's a good point. It's probably worth doing in a separate PR though.

re: Load more button
Good catch! Right now there's a mismatch between the threshold in the controller and view template
https://github.com/department-of-veterans-affairs/diffusion-marketplace/blob/66799e467dc18bfbe4ddf3fa29b737f6332a6ea6/app/controllers/page_controller.rb#L106
https://github.com/department-of-veterans-affairs/diffusion-marketplace/blob/66799e467dc18bfbe4ddf3fa29b737f6332a6ea6/app/views/page/show.html.erb#L202

Setting them both to 4 is fine for now. I noticed in the Figma designs that there are different thresholds for the homepage and the News / Events page, so we should discuss options with Becca.

File: app/views/page/_news_items_list.html.erb [hunk 2]
```diff
@@ -6,15 +6,27 @@
...
             <div id="<%= news_item.id %>" class="news-item-description usa-prose usa-card__body">
-                <p><%= news_item&.text&.html_safe %></p>
                 <% if news_item.published_date.present? && news_item.authors.present? %>
                     <p>Published on <%= news_item&.published_date&.strftime("%B %e, %Y") %> by <%= news_item&.authors %></p>
                 <% elsif news_item.published_date.present? %>
                     <p>Published on <%= news_item&.published_date&.strftime("%B %e, %Y") %></p>
                 <% elsif news_item.authors.present? %>
                     <p>Published by <%= news_item&.authors %></p>
                 <% end %>
+                <div><%= news_item&.text&.html_safe %></div>
```
Comment (PhilipDeFraties) [inline app/views/page/_news_items_list.html.erb:29]: moved the text body to the bottom to reflect figma mockup

**Review by camillevilla — CHANGES_REQUESTED** [review]
👍 Code for this component extension is working well. Snappy little image preview is great too. 

We need to constrain the images to the card area size. e.g. if I upload images of different sizes, they should all get cropped / masked to a similar size. Let's take a look at practice cards to see if there's a pattern we can borrow there. We don't need to provide anything too fancy like cropping tools within the application. Generally we expect content creators to crop and resize images to fit the provided image ratio of 1:2 length:width.

### News Components
Screenshot 2023-07-12 at 2 42 23 PM

### Practice cards
Screenshot 2023-07-12 at 2 42 27 PM

**Review by camillevilla — APPROVED** [review]
✨🖼️ 
Screenshot 2023-07-13 at 2 57 16 PM

**Pull request merged**